Use the Insecure setting added in PR #4 when creating the API Client

cnelson opened the pull request:
One more PR from me today :)

While testing the new insecure setting, I noticed it doesn't seem to work, as the API Client wasn't looking at that setting, so API calls would cause a failure before fly is ever executed.

robdimsdale commented:
Ah, ok.

Can we get a test to cover the insecure behavior in the `pipelines_test.go`? Otherwise this looks great and ready to merge.

cnelson commented:
Will do, I'll add a test;  

Also wondering if  https://github.com/robdimsdale/concourse-pipeline-resource/blob/master/fly/fly.go#L47-L51 should come out.  I think @o-orand was setting the DefaultClient as the ApiClient used to use that, but since I added basic auth, it has it's own Client.

cnelson commented:
Added that test, but not sure if there's a better or more robust way of going about it than adding that getter.  Feedback appreciated.

robdimsdale commented:
Yeah I agree that test is a bit out of place, and doesn't really test the functionality that we expect - i.e. that it skips SSL verify. Maybe we should just back out that test? It would be fairly easy to end-to-end test, but then that would require a concourse instance running with an invalid SSL cert.

Let's revert the commit that adds the test, and move forward without coverage for now. We may extract a client package in the future, in which case we can test its behavior in isolation, including the `InsecureSkipVerify` config.

One other thing, we should make the new argument to the `NewClient` method a `bool`, not `string`; this change was made for the `fly` package after #4 was merged.

Thanks for sticking with me on this as we discover what makes the most sense - much appreciated!

cnelson commented:
Dropped the test commit, and switched the NewClient() to take a `bool` for the `insecure` option.

> Thanks for sticking with me on this as we discover what makes the most sense - much appreciated!

No problem at all! Thanks for being so responsive, I'd have way fewer forks if every project owner was so prompt 👍

If you are willing to add [stunnel](https://www.stunnel.org/index.html) as a test dependency for SSL tests, I have some code from another project I could contribute which would put `stunnel` in front of the `$TARGET` with a specific configuration. 

With that, more acceptance_tests can be written that easily validate all TLS scenarios.

robdimsdale commented:
Looks great, thanks. Merged.

I like the idea of easily testing the various TLS configurations, so `stunnel` might be an option. I do think it makes more sense to extract the transport layer-logic into a separate package and test the TLS permutations there against an in-memory webserver (like gomega's [`ghttp`](https://onsi.github.io/gomega/#ghttp-testing-http-clients)) and I'm not particularly worried about the test coverage just yet.

One of my goals for this project is to migrate away from exec-ing out to the `fly` binary and instead use the [go-concourse](https://github.com/concourse/go-concourse) library - this would likely mean using a common `http.Transport` everywhere, and then I think the value of testing the extracted transport package increases.

Anyway, thanks again for the PR. I haven't tested this so please let me know if there are any issues

cnelson commented:
After a quick test it LGTM.  Thanks again!
